openedx/core/djangoapps/schedules/config.py

```diff
@@ -19,3 +24,48 @@
     'course_update_show_unsubscribe',
     __name__
 )
+
+# This experiment waffle is supporting an A/B test we are running on sending course updates from an external service,
+# rather than through platform and ACE. See ticket AA-661 for more information.
+# Don't use this flag directly, instead use the `set_up_external_course_updates` and `query_external_updates` methods
+# below. We save this flag decision at enrollment time and don't change it even if the flag changes. So you can't just
+# directly look at flag result.
+_EXTERNAL_COURSE_UPDATES_EXPERIMENT_ID = 18
+_EXTERNAL_COURSE_UPDATES_FLAG = ExperimentWaffleFlag(WAFFLE_FLAG_NAMESPACE, 'external_updates', __name__,
+                                                     experiment_id=_EXTERNAL_COURSE_UPDATES_EXPERIMENT_ID,
+                                                     use_course_aware_bucketing=False)
+
+
+def set_up_external_course_updates(user, course_key):
```

Comment:
Imma talk about the approach here a little bit. When a user enrolls in a course, we call this method, which checks the waffle and then stores the value in our ExperimentData table.

We store the value so that it won't change over the enrollment lifetime, even if waffle does change.

If a user gets bucketed into the new experience, they will get that experience for all new courses going forward. But stay with old experience on existing courses before they got bucketed.

Because users can be enrolled by third parties (like course staff), we check the user here and make sure we only bucket users that are enrolling themselves. This is because django waffle flags can only work on the request user, and it felt grosser to fake a request user than to just zero-bucket those cases.

Reply:
would it be clearer to name this method something like bucket_whether_user_sees_external_course_updates. 
(though the term external_course_updates is kind of confusing too, can you explain in the comment what that is?)

also you mean negative one bucket rather than zero, right?

Reply:
> would it be clearer to name this method something like bucket_whether_user_sees_external_course_updates.

Naming is hard, yes. I wanted something (I went with `set_up`) that indicates we are not just bucketing, but bucketing forever. I'm not wedded to my name, though.

>  (though the term external_course_updates is kind of confusing too, can you explain in the comment what that is?)

I explain the phrase and experiment above, near the waffle flag definitions. Are you suggesting I also explain it in this method's docstring as well because it might be missed above, or did you mean here in the PR?

> also you mean negative one bucket rather than zero, right?

Yes correct, we return a -1 (per Tim's request to separate those) for bad-user requests. But the effect is the same as a 0 bucket. They get the control/traditional experience.

Reply:
maybe set_up_whether_user_sees_external_course_updates?

Reply:
I don't love repeating information from the argument list in the function name. But I do get your point that this method sounds like it sets up the whole experiment altogether.

Maybe `set_up_external_updates_for_enrollment(user, course)`?

Reply:
maybe initialize_external_course_updates_experiment_data?

Reply:
Can you expand on what you don't like about the name (I've currently changed it to `set_up_external_updates_for_enrollment`, but it's super easy to change again)?

I thought you were coming at the angle of it should clarify that it's for a specific user/course. But your last name doesn't do that. Is it that you feel the phrase `set_up` is ambiguous?

Reply:
i think it was more than you're setting up the data that says whether a user should see the external course updates, not actually setting up the external course updates

Reply:
Yeah, fair. I am going to say that the name is close enough now (with the addition of `_for_enrollment`) and the code temporary enough (I hope...), that I'll just leave it.